### PR TITLE
Created Dart Debug Extension request & response.

### PR DESCRIPTION
**Pull request by pisimulation**

### CODE REVIEW

File: dwds/lib/data/extension_request.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+
+part 'extension_request.g.dart';
+
+/// A request to run a command on Dart Debug Extension.
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: [nit] A request to run a command in the* Dart Debug Extension.

File: dwds/lib/data/extension_request.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  static Serializer<ExtensionRequest> get serializer =>
+      _$extensionRequestSerializer;
+
+  factory ExtensionRequest([Function(ExtensionRequestBuilder) updates]) =
+      _$ExtensionRequest;
+
+  ExtensionRequest._();
+
+  /// Identifies which completer sends this request.
+  int get completerId;
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: Let's just simply name this id. The doc comment can say something like:

"/// Used to associate a request with an ExtensionResponse"

File: dwds/lib/data/extension_request.dart [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  static Serializer<ExtensionResponse> get serializer =>
+      _$extensionResponseSerializer;
+
+  factory ExtensionResponse([Function(ExtensionResponseBuilder) updates]) =
+      _$ExtensionResponse;
+
+  ExtensionResponse._();
+
+  /// Identifies which completer to receive the response.
+  int get completerId;
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: See note about ID above.

File: dwds/lib/data/extension_request.dart [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  bool get success;
+
+  String get result;
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: Let's put a doc comment stating that this is a json encoded payload.

File: dwds/lib/data/extension_request.dart [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      _$ExtensionRequest;
+
+  ExtensionRequest._();
+
+  /// Identifies which completer sends this request.
+  int get completerId;
+
+  String get command;
+
+  @nullable
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: Should this be nullable or do we always send an empty map?

Comment (pisimulation) [reply dwds/lib/data/extension_request.dart]: Good point. We always send an empty map if there is no parameter.

File: dwds/lib/data/extension_request.dart [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    implements Built<ExtensionResponse, ExtensionResponseBuilder> {
+  static Serializer<ExtensionResponse> get serializer =>
+      _$extensionResponseSerializer;
+
+  factory ExtensionResponse([Function(ExtensionResponseBuilder) updates]) =
+      _$ExtensionResponse;
+
+  ExtensionResponse._();
+
+  /// Used to associate a response with an ExtensionRequest.
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: [nit] Brackets around ExtensionRequest

File: dwds/lib/data/extension_request.dart [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    implements Built<ExtensionRequest, ExtensionRequestBuilder> {
+  static Serializer<ExtensionRequest> get serializer =>
+      _$extensionRequestSerializer;
+
+  factory ExtensionRequest([Function(ExtensionRequestBuilder) updates]) =
+      _$ExtensionRequest;
+
+  ExtensionRequest._();
+
+  /// Used to associate a request with an ExtensionResponse.
```
Comment (grouma) [inline dwds/lib/data/extension_request.dart]: [nit] Brackets around ExtensionResponse.

**Pull request merged**